codec-http2/src/main/java/io/netty/handler/codec/http2/Http2ConnectionHandler.java

```diff
@@ -695,7 +697,9 @@ public void operationComplete(ChannelFuture future) throws Exception {
 
             return future;
         } catch (Throwable cause) { // Make sure to catch Throwable because we are doing a retain() in this method.
-            debugData.release();
+            if (release) {
```

Comment:
![BLOCKER](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-blocker.png) Change this condition so that it does not always evaluate to "false" [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS2583)